plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -256,6 +268,30 @@ else if (upperBound != null) {
         return Domain.create(ValueSet.ofRanges(statisticsRange), containsNulls);
     }
 
+    static boolean partitionMatchesConstraint(
+            Set<IcebergColumnHandle> identityPartitionColumns,
+            Map<Integer, Optional<String>> partitionKeys,
+            Constraint constraint)
+    {
+        verify(constraint.getSummary().isAll());
```

Comment:
add a comment
// We use Constraint just to pass functional predicate here from DistributedExecutionPlanner